src/pgnViewer.ts

```diff
@@ -1,49 +1,82 @@
 import { Api as CgApi } from 'chessground/api';
-import { makeSquare, opposite } from 'chessops';
+import { makeSquare, opposite, Color } from 'chessops';
 import translator from './translation';
 import { GoTo, InitialOrMove, Opts, Translate } from './interfaces';
 import { Config as CgConfig } from 'chessground/config';
 import { uciToMove } from 'chessground/util';
 import { Path } from './path';
 import { AnyNode, Game, isMoveData } from './game';
 import { makeGame } from './pgn';
+import { parsePgn } from 'chessops/pgn';
 
 export default class PgnViewer {
-  game: Game;
+  games: Game[];
   path: Path;
   translate: Translate;
   ground?: CgApi;
   div?: HTMLElement;
   flipped = false;
   pane = 'board';
   autoScrollRequested = false;
+  selectedGame = 0;
 
   constructor(
     readonly opts: Opts,
     readonly redraw: () => void,
   ) {
-    this.game = makeGame(opts.pgn, opts.lichess);
-    opts.orientation = opts.orientation || this.game.metadata.orientation;
+    let games = parsePgn(opts.pgn);
+    if (games.length === 0) games = parsePgn('*');
+    this.games = games.map(game => makeGame(game, opts.lichess));
     this.translate = translator(opts.translate);
-    this.path = this.game.pathAtMainlinePly(opts.initialPly);
+    this.path = this.game.pathAtMainlinePly(this.opts.initialPly);
   }
 
+  get game() {
+    return this.games[this.selectedGame];
+  }
+  selectIndex = (i: number) => {
+    this.selectedGame = Math.max(0, Math.min(this.games.length - 1, i));
+    this.path = Path.root;
+    this.redrawGround();
+    this.redraw();
+  };
+
   curNode = (): AnyNode => this.game.nodeAt(this.path) || this.game.moves;
   curData = (): InitialOrMove => this.game.dataAt(this.path) || this.game.initial;
 
   goTo = (to: GoTo, focus = true) => {
-    const path =
-      to == 'first'
-        ? Path.root
-        : to == 'prev'
-        ? this.path.init()
-        : to == 'next'
-        ? this.game.nodeAt(this.path)?.children[0]?.data.path
-        : this.game.pathAtMainlinePly('last');
+    let index = this.selectedGame;
+    let path = this.path;
+
+    if (to === 'first') path = Path.root;
+    if (to === 'last') path = this.game.pathAtMainlinePly('last');
+    if (to === 'prev') path = this.path.init();
+    if (to === 'next') path = this.game.nodeAt(this.path)?.children?.[0]?.data?.path ?? path;
+
+    if (this.path.path === path.path) {
+      if (to === 'first') index = 0;
+      if (to === 'last') index = this.games.length - 1;
+      if (to === 'prev') index--;
+      if (to === 'next') index++;
+      if (index < 0) index = 0;
+      if (index >= this.games.length) index = this.games.length - 1;
+    }
+
+    if (index !== this.selectedGame) {
+      this.selectIndex(index);
+      if (to === 'first') path = Path.root;
+      if (to === 'last') path = this.game.pathAtMainlinePly('last');
```

Comment:
`first` and `last` here are weird. They'll jump to the first or last game, but only if we're at the first or last ply of the current game. That doesn't seem intuitive.